src/modules/taskStore.js

```diff
@@ -0,0 +1,75 @@
+import Storage from './storage.js';
+import Task from './task.js';
+
+const ITEM_STORAGE = 'toDoTasks';
+/**
+ * Class to Add, Remove and Get tasks from Storage from
+ */
+export default class TaskStore {
+  tasksCollection = [];
+
+  taskStorage;
+
+  constructor() { this.taskStorage = new Storage(ITEM_STORAGE); }
+
+  add= (description) => {
+    this.tasksCollection = this.taskStorage.getItemStorage();
+    const id = (this.tasksCollection !== null) ? this.tasksCollection.length + 1 : 1;
+    const objectTask = new Task(id, false, description);
+    this.taskStorage.addItemStorage(objectTask);
+  }
+
+  remove=(id) => {
+    if (id > 0) {
+      this.tasksCollection = this.taskStorage.getItemStorage();
+      const resultCollection = this.tasksCollection.filter((t) => t.index !== parseInt(id, 10));
+      let newId = 1;
+      // reindexation
+      resultCollection.forEach((element) => {
+        element.index = newId;
+        newId += 1;
+      });
+      this.taskStorage.updateItemStorage(resultCollection);
+    }
+  }
+
+  clearCompleted=() => {
+    this.tasksCollection = this.taskStorage.getItemStorage();
+    const resultCollection = this.tasksCollection.filter((t) => t.completed === false);
+    let newId = 1;
+    // reindexation
+    resultCollection.forEach((element) => {
+      element.index = newId;
+      newId += 1;
+    });
+    this.taskStorage.updateItemStorage(resultCollection);
+  }
```

Comment:
- Hereafter update of collection, the update-related function has not been called and that is why on refresh it is not refreshing the checkout button. You can consider it to update.

![Screenshot from 2022-04-05 20-00-14](https://user-images.githubusercontent.com/78268827/161778066-b961b564-2825-49da-910f-09a40f529c60.png)